### DISCUSSION TITLE
treelog log hyperlink

### DISCUSSION THREAD

**Discussion by SteinStoter** [evalf/nutils#865] [Q&A] [answered]
I'd like to log a link, but the treelog html logger appears to escape the necessary characters. How do I work around that?
`treelog.info("Example link:   <a href=\"https://mefd.pages.tue.nl/home/\">MEFD webpage</a>  ")`
produces as html output:
`<div class="item" data-loglevel="1">Example link: &lt;a href=&quot;https://mefd.pages.tue.nl/home/&quot;&gt;MEFD webpage&lt;/a&gt; </div>`

**Comment by SteinStoter** [comment, marked as answer]
I managed to do this by exposing the `HtmlLog` instance to which I am logging. This has a `write` function, with an `escape` kwarg that you can set to `False`.

**Reply by gertjanvanzwieten** [reply]
This is indeed the correct answer, you need a specific handle to the `HtmlLog` object to access the low access write functionality. To avoid this I think would be open to change the behaviour of the html logger to detect URIs in logged text and turn them into clickable links automatically, so that escaping of manually placed html tags is no longer an issue. Feel free to make that an issue for this in the treelog project if that sounds useful!
